src/storage/index/IndexExecutor.h

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef STORAGE_INDEXEXECUTOR_H
+#define STORAGE_INDEXEXECUTOR_H
+
+#include "stats/Stats.h"
+#include "storage/index/IndexPolicyMaker.h"
+
+namespace nebula {
+namespace storage {
+
+template<typename RESP>
+class IndexExecutor : public BaseProcessor<RESP>
+                    , public IndexPolicyMaker {
+public:
+    virtual ~IndexExecutor() = default;
+
+protected:
+    explicit IndexExecutor(kvstore::KVStore* kvstore,
+                           meta::SchemaManager* schemaMan,
+                           meta::IndexManager* indexMan,
+                           stats::Stats* stats,
+                           VertexCache* cache,
+                           bool isEdgeIndex = false)
+        : BaseProcessor<RESP>(kvstore, schemaMan, stats)
+        , IndexPolicyMaker(schemaMan, indexMan)
+        , vertexCache_(cache)
+        , isEdgeIndex_(isEdgeIndex) {}
+
+    void putResultCodes(cpp2::ErrorCode code, const std::vector<PartitionID>& parts) {
+        for (auto& p : parts) {
+            this->pushResultCode(code, p);
+        }
+        this->onFinished();
+    }
+
+    cpp2::ErrorCode prepareRequest(const cpp2::LookUpIndexRequest &req);
+
+    /**
+     * Details Prepare the index scan. logic as below :
+     *         1, Trigger policy generator
+     *         2, Build prefix string for first n columns of index.
+     *         3, Collect information needed for index scanning,
+     **/
+    cpp2::ErrorCode buildExecutionPlan(const std::string& filter);
+
+    /**
+     * Details Scan index part as one by one.
+     **/
+    kvstore::ResultCode executeExecutionPlan(PartitionID part);
```

Comment:
Sorry to say, the execution plan was representative the behavior of `Query Engine` do you mind use another name ?

Reply:
> Sorry to say, the execution plan was representative the behavior of `Query Engine` do you mind use another name ?

Sounds reasonable，any Suggestions?

Reply:
I think its fine here.